echopype/consolidate/api.py

```diff
@@ -55,84 +60,123 @@ def swap_dims_channel_frequency(ds: Union[xr.Dataset, str, pathlib.Path]) -> xr.
         )
 
 
+@add_processing_level("L2A")
 def add_depth(
     ds: Union[xr.Dataset, str, pathlib.Path],
-    depth_offset: float = 0,
-    tilt: float = 0,
+    echodata: Optional[Union[EchoData, str, pathlib.Path]] = None,
+    depth_offset: float = 0.0,
+    tilt: float = 0.0,
     downward: bool = True,
+    use_platform_vertical_offsets: bool = False,
+    use_platform_angles: bool = False,
+    use_beam_angles: bool = False,
 ) -> xr.Dataset:
     """
-    Create a depth data variable based on data in Sv dataset.
-
-    The depth is generated based on whether the transducers are mounted vertically
-    or with a polar angle to vertical, and whether the transducers were pointed
-    up or down.
+    Create a depth data variable based on data in Sv dataset, Echodata object, and/or
+    user input depth offset and tilt data.
 
     Parameters
     ----------
     ds : xr.Dataset or str or pathlib.Path
-        Source Sv dataset or path to a file containing the Source Sv dataset
-        to which a depth variable will be added.
-        Must contain `echo_range`.
-    depth_offset : float
+        Source Sv dataset to which a depth variable will be added.
+    echodata : EchoData or str or pathlib.Path, optional, default `None`
+        `EchoData` object from which the `Sv` dataset originated.
+    depth_offset : float, default `0.0`
         Offset along the vertical (depth) dimension to account for actual transducer
         position in water, since `echo_range` is counted from transducer surface.
-        Default is 0.
-    tilt : float
-        Transducer tilt angle [degree].
-        Default is 0 (transducer mounted vertically).
-    downward : bool
-        Whether or not the transducers point downward.
-        Default to True.
+        Will only be used if `use_platform_vertical_offsets=False`.
+    tilt : float, default `0.0`
+        Transducer tilt angle [degree]. 0 corresponds to a transducer pointing vertically.
+        Will only be  used if both `use_platform_angles=False` and `use_beam_angles=False`.
+    downward : bool, default `True`
+        The transducers point downward.
+    use_platform_vertical_offsets: bool, default `False`
+        If True, use Echodata Platform group vertical offset values to compute transducer depth.
+        Currently only implemented for EK60/EK80 sonar models.
+    use_platform_angles: bool, `False`
+        If True, use Echodata Platform group angle values to compute `echo_range` scaling values.
+        Currently only implemented for EK60/EK80 sonar models.
+        Cannot be used in tandem with Echodata Beam group angles.
+    use_beam_angles: bool `False`
+        If True, use Echodata Beam group angle values to compute `echo_range` scaling values.
+        Currently only implemented for EK60/EK80 sonar models.
+        Cannot be used in tandem with Echodata Platform group angles.
```

Comment:
is the following correct?
```suggestion
    use_platform_angles: bool, `False`
        If True, use Echodata Platform group angle values to compute `echo_range` scaling values.
        Currently only implemented for EK60/EK80 sonar models.
        In the current implementation cannot be used in tandem with `use_beam_angles`.
    use_beam_angles: bool `False`
        If True, use Echodata Beam group angle values to compute `echo_range` scaling values.
        Currently only implemented for EK60/EK80 sonar models.
        In the current implementation cannot be used in tandem with `use_platform_angles`.
```

Reply:
Yup I'll add this